modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -0,0 +1,230 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.util;
+
+import static org.mule.MessageExchangePattern.REQUEST_RESPONSE;
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.DefaultMuleEvent;
+import org.mule.DefaultMuleMessage;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.construct.FlowConstruct;
+import org.mule.extension.annotations.param.Optional;
+import org.mule.extension.introspection.Configuration;
+import org.mule.extension.introspection.Described;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.runtime.ConfigurationInstanceProvider;
+import org.mule.extension.runtime.OperationContext;
+import org.mule.module.extension.internal.runtime.ConfigurationObjectBuilder;
+import org.mule.module.extension.internal.runtime.DynamicConfigurationInstanceProvider;
+import org.mule.module.extension.internal.runtime.OperationContextAdapter;
+import org.mule.module.extension.internal.runtime.StaticConfigurationInstanceProvider;
+import org.mule.module.extension.internal.runtime.resolver.ResolverSet;
+import org.mule.module.extension.internal.runtime.resolver.ValueResolver;
+import org.mule.util.TemplateParser;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.LinkedHashMultiset;
+import com.google.common.collect.Multiset;
+import com.google.common.collect.Multisets;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+
+public final class MuleExtensionUtils
+{
+
+    public static void checkNullOrRepeatedNames(Collection<? extends Described> describedCollection, String describedEntityName)
+    {
+        Set<String> repeatedNames = collectRepeatedNames(describedCollection, describedEntityName);
+
+        if (!repeatedNames.isEmpty())
+        {
+            throw new IllegalArgumentException(
+                    String.format("The following %s were declared multiple times: [%s]",
+                                  describedEntityName,
+                                  Joiner.on(", ").join(repeatedNames))
+            );
+        }
+    }
+
+    /**
+     * Verifies that no operation has the same name as a configuration. This
+     * method assumes that the configurations and operations provided have
+     * already been verified through {@link #checkNullOrRepeatedNames(java.util.Collection, String)}
+     * which means that name clashes can only occur against each other and not within the
+     * inner elements of each collection
+     */
```

Comment:
Javadoc must indicate that an IllegalArgumentException is thrown when there is a name clash